matchbox-engine/src/main/java/org/hl7/fhir/r5/context/BaseWorkerContext.java

```diff
@@ -136,2285 +136,2267 @@ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWIS
+	public <T extends Resource> List<T> fetchResourcesByType(Class<T> class_) {
+
+		List<T> res = new ArrayList<>();
+
+		synchronized (lock) {
+
+			if (class_ == Resource.class || class_ == DomainResource.class || class_ == CanonicalResource.class || class_ == null) {
+				res.addAll((List<T>) structures.getList());
+				res.addAll((List<T>) guides.getList());
+				res.addAll((List<T>) capstmts.getList());
+				res.addAll((List<T>) measures.getList());
+				res.addAll((List<T>) libraries.getList());
+				res.addAll((List<T>) valueSets.getList());
+				res.addAll((List<T>) codeSystems.getList());
+				res.addAll((List<T>) operations.getList());
+				res.addAll((List<T>) searchParameters.getList());
+				res.addAll((List<T>) plans.getList());
+				res.addAll((List<T>) maps.getList());
+				res.addAll((List<T>) transforms.getList());
+				res.addAll((List<T>) questionnaires.getList());
+				res.addAll((List<T>) systems.getList());
+				res.addAll((List<T>) actors.getList());
+				res.addAll((List<T>) requirements.getList());
+			} else if (class_ == ImplementationGuide.class) {
+				res.addAll((List<T>) guides.getList());
+			} else if (class_ == CapabilityStatement.class) {
+				res.addAll((List<T>) capstmts.getList());
+			} else if (class_ == Measure.class) {
+				res.addAll((List<T>) measures.getList());
+			} else if (class_ == Library.class) {
+				res.addAll((List<T>) libraries.getList());
+			} else if (class_ == StructureDefinition.class) {
+				res.addAll((List<T>) structures.getList());
+			} else if (class_ == StructureMap.class) {
+				res.addAll((List<T>) transforms.getList());
+			} else if (class_ == ValueSet.class) {
+				res.addAll((List<T>) valueSets.getList());
+			} else if (class_ == CodeSystem.class) {
+				res.addAll((List<T>) codeSystems.getList());
+			} else if (class_ == NamingSystem.class) {
+				res.addAll((List<T>) systems.getList());
+			} else if (class_ == ActorDefinition.class) {
+				res.addAll((List<T>) actors.getList());
+			} else if (class_ == Requirements.class) {
+				res.addAll((List<T>) requirements.getList());
+			} else if (class_ == ConceptMap.class) {
+				res.addAll((List<T>) maps.getList());
+			} else if (class_ == PlanDefinition.class) {
+				res.addAll((List<T>) plans.getList());
+			} else if (class_ == OperationDefinition.class) {
+				res.addAll((List<T>) operations.getList());
+			} else if (class_ == Questionnaire.class) {
+				res.addAll((List<T>) questionnaires.getList());
+			} else if (class_ == SearchParameter.class) {
+				res.addAll((List<T>) searchParameters.getList());
+			}
+		}
+		return res;
+	}
+
+	private Set<String> notCanonical = new HashSet<String>();
+
+	protected IWorkerContextManager.IPackageLoadingTracker packageTracker;
+
+	@Override
+	public Resource fetchResourceById(String type, String uri) {
+		synchronized (lock) {
+			String[] parts = uri.split("\\/");
+			if (!Utilities.noString(type) && parts.length == 1) {
+				if (allResourcesById.containsKey(type)) {
```

Comment:
if (!Utilities.noString(type) && parts.length == 1) {
        if (allResourcesById.containsKey(type)) {
          // matchbox prevent NP
          ResourceProxy proxy = allResourcesById.get(type).get(parts[0]);
          return proxy != null ?  proxy.getResource() : null;
        } else {
          return null;
        }

Reply:
Thanks, fixed in 8dad8bdb87d330fe202cef3c9f8de1264d029cd2